concourse/steps/replicate_secrets.py

```diff
@@ -143,55 +144,79 @@ def _put_secret(
 
 
 def replicate_secrets(
+    cluster_kubeconfig: dict, # kubeconfig with access to secrets-server-namespace
+    concourse_team_name: str,
     cfg_factory: model.ConfigFactory,
-    cfg_set: model.ConfigurationSet,
-    kubeconfig: typing.Dict,
-    secret_key: str,
-    secret_cipher_algorithm: str,
-    future_secrets: typing.Dict[str, str],
-    team_name: str,
-    target_secret_name: str,
-    target_secret_namespace: str,
-    target_secret_cfg_name: str,
+    replication_target_config: model.config_repo.ReplicationTargetConfig,
 ):
-    logger.info(f'replicating replication cfg set {cfg_set.name()}')
-
-    # force cfg_set serialiser to include referenced cfg_sets
-    cfg_sets = list(cfg_set._cfg_elements('cfg_set')) + [cfg_set]
-
-    for cfg_set in cfg_sets:
-        logger.info(f'config subset {cfg_set.name()=} with keys')
-        for cfg_mapping in [cfg_set._cfg_mappings()]:
-            for cfg_type_name, _ in cfg_mapping:
-                pprint.pprint(
-                    {cfg_type_name: cfg_set._cfg_element_names(cfg_type_name=cfg_type_name)}
-                )
+    for mapping in replication_target_config.replication_mappings:
+        cfg_set = cfg_factory.cfg_set(mapping.cfg_set)
+        # force cfg_set serialiser to include referenced cfg_sets
+        cfg_sets = list(cfg_set._cfg_elements('cfg_set')) + [cfg_set]
+        if isinstance(target := mapping.target, model.config_repo.KubernetesSecretTarget):
```

Comment:
considering there will be multiple cases, rather pull this codeblock into a function.

also, would it not make sense to invert the iteration, like so:

```
for cfg_set in cfg_sets:
  serialised_cfg = <code to serialise cfg-set>
  for target in targets:
    if <is-k8s-tgt>:
     put_secret(serialised_cfg)
   elif <is-other-tgt>:
     put_secret_otherwise(serialised_cfg)
```

wdyt?